test/spec/app_spec.rb

```diff
@@ -830,6 +788,150 @@
+      @driver.find_element(:css,'.media-actions__edit').click
+      sleep 3
+      fill_field('.ReactTags__tagInput input', "TAG")
+      @driver.action.send_keys("\n").perform      
+      sleep 1
+      fill_field('.ReactTags__tagInput input', "TAG2")
+      @driver.action.send_keys("\n").perform
+      expect(page.tags.length == 2).to be(true)
+      list = @driver.find_elements(:class, "ReactTags__remove")
+      list[1].click
+      sleep 3
+      expect(page.tags.length == 1).to be(true)
+    end
+
+    it "should show 'manage team' link only to team owners", bin3: true do
+      user = api_register_and_login_with_email
+      team = request_api 'team', { name: "team#{Time.now.to_i}", email: user.email, slug: "team#{Time.now.to_i}" }
+      user2 = api_register_and_login_with_email
+      page = TeamsPage.new(config: @config, driver: @driver).load
+      page.ask_join_team(subdomain: team.slug)
+      @wait.until {
+        expect(@driver.find_element(:class, "message").nil?).to be(false)
+      }
+      api_logout
+      @driver = new_driver(webdriver_url,browser_capabilities)
+      page = Page.new(config: @config, driver: @driver)
+      page.go(@config['api_path'] + '/test/session?email='+user.email)
+      #As the group creator, go to the members page and approve the joining request.
+      page = TeamsPage.new(config: @config, driver: @driver).load
+          .approve_join_team(subdomain: team.slug)
+      api_logout
+
+      @driver = new_driver(webdriver_url,browser_capabilities)
+      page = Page.new(config: @config, driver: @driver)
+      page.go(@config['api_path'] + '/test/session?email='+user2.email)
+      page = TeamsPage.new(config: @config, driver: @driver).load
+      sleep 3
+      @driver.find_element(:class,"header-actions__menu-toggle").click
+      sleep 3
+      expect(@driver.page_source.include?('Manage team')).to be(false)
+    end
+
+
+    it "should show 'edit project' link only to users with 'update project' permission", bin3: true do
+      utp = api_create_team_and_project()
+      user2 = api_register_and_login_with_email
+      page = TeamsPage.new(config: @config, driver: @driver).load
+      page.ask_join_team(subdomain: utp[:team].slug)
+      sleep 3        
+      api_logout
+      @driver = new_driver(webdriver_url,browser_capabilities)
+      page = Page.new(config: @config, driver: @driver)
+      page.go(@config['api_path'] + '/test/session?email='+utp[:user].email)
+      #As the group creator, go to the members page and approve the joining request.
+      page = TeamsPage.new(config: @config, driver: @driver).load
+          .approve_join_team(subdomain: utp[:team].slug)
+      go(@config['self_url'] + '/'+utp[:team].slug+'/project/'+utp[:project].dbid.to_s)
+      sleep 3
+      @driver.find_element(:class,"header-actions__menu-toggle").click
+      sleep 3
+      expect(@driver.page_source.include?('Edit project')).to be(true)
+      api_logout
```

Comment:
This whole process of adding a user to a team should be handled by the test API, to avoid repetition between tests. Please tell us what this test endpoint should do (for example, "given a team, create a user under that team with a given role") and someone can implement that on the backend.